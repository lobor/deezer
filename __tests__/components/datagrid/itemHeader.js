import React from 'react';
import ItemHeader from 'components/datagrid/itemHeader';
import renderer from 'react-test-renderer';

describe('test components itemHeader', () => {
  it('test render', () => {
    const component = renderer.create(
      <ItemHeader title="Toto" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('test render width filter', () => {
    const filter = jest.mock();
    const sort = jest.mock();
    const component = renderer.create(
      <ItemHeader title="Toto" filter={filter} sort={sort} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('test render width sort', () => {
    const component = renderer.create(
      <ItemHeader title="Toto" />
    );
    let tree = component.toJSON();
    tree.props.sort = jest.fn();
    tree.props.onClick({ target: { tagName: 'TD' } });
    tree.props.onClick({ target: { tagName: 'TD' } });
    tree.props.onClick({ target: { tagName: 'TD' } });
  })
})
