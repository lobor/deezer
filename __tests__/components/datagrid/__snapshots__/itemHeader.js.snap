// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test components itemHeader test render 1`] = `
<th
  onClick={[Function]}
  style={Object {}}
>
  <span
    className={null}
  >
    Toto
  </span>
</th>
`;

exports[`test components itemHeader test render width filter 1`] = `
<th
  onClick={[Function]}
  style={Object {}}
>
  <span
    className={null}
  >
    Toto
    <label>
      <input
        className="input"
        name=""
        onChange={
          Object {
            "addMatchers": [Function],
            "autoMockOff": [Function],
            "autoMockOn": [Function],
            "clearAllMocks": [Function],
            "clearAllTimers": [Function],
            "deepUnmock": [Function],
            "disableAutomock": [Function],
            "doMock": [Function],
            "dontMock": [Function],
            "enableAutomock": [Function],
            "fn": [Function],
            "genMockFn": [Function],
            "genMockFromModule": [Function],
            "genMockFunction": [Function],
            "isMockFunction": [Function],
            "mock": [Function],
            "requireActual": [Function],
            "requireMock": [Function],
            "resetAllMocks": [Function],
            "resetModuleRegistry": [Function],
            "resetModules": [Function],
            "restoreAllMocks": [Function],
            "runAllImmediates": [Function],
            "runAllTicks": [Function],
            "runAllTimers": [Function],
            "runOnlyPendingTimers": [Function],
            "runTimersToTime": [Function],
            "setMock": [Function],
            "setTimeout": [Function],
            "spyOn": [Function],
            "unmock": [Function],
            "useFakeTimers": [Function],
            "useRealTimers": [Function],
          }
        }
        placeholder="filter"
        type="text"
      />
    </label>
  </span>
</th>
`;
